input/otlp/metadata.go

```diff
@@ -374,12 +377,31 @@ func translateResourceMetadata(resource pcommon.Resource, out *modelpb.APMEvent)
 		// agent.version is a required field.
 		out.Agent.Version = "unknown"
 	}
-	if out.GetService().GetLanguage().GetName() != "" {
-		if out.Agent == nil {
-			out.Agent = modelpb.AgentFromVTPool()
+
+	distroName, distroNameSet := resource.Attributes().Get("telemetry.distro.name")
+	distroVersion, distroVersionSet := resource.Attributes().Get("telemetry.distro.version")
+
+	if distroNameSet && distroName.Str() != "" {
+		agentLang := "unknown"
+		if out.GetService().GetLanguage().GetName() != "" {
+			agentLang = out.GetService().GetLanguage().GetName()
+		}
+
+		out.Agent.Name = fmt.Sprintf("%s/%s/%s", out.Agent.Name, agentLang, distroName.Str())
+
+		//we intentionally do not want to fallback to the Otel SDK version if we have a distro name, this would only cause confusion
+		out.Agent.Version = "unknown"
```

Comment:
@AlexanderWert @estolfo do you agree that it's better to use `unknown` as `agent.version` instead of the sdk-version if a distro-name was found but no distro-version?

I'm thinking that the ambiguity that in that case `agent.version` could be both the distro version or the sdk-version would cause a lot of confusion if both follow similar versioning schemes

Reply:
Agreed on in the agents-weekly meeting.